### DISCUSSION TITLE
Select name , how can i get the value of the select onChange when i have more than one

### DISCUSSION THREAD

**Discussion by itsosmx** [JedWatson/react-select#4763] [Q&A]
So i have more than one input everyone have its name that make everything goes where i want , now select only return the selected value without anything make me know what is this value belong to I try to use name attribute  but nothing change for me 
else its hard to make a handle for every input

**Comment by Rall3n** [comment]
Hello @itsosmx,

I hope you don't mind me converting your issue to an discussion, as this is the right place for questions such as yours.

The `onChange` callback prop should accept 2 arguments: the changed value as the first argument, and a meta object as the second argument. This meta object contains the value of the `name` prop as the property `name`.

A reusable callback could look like this:
```jsx
const callback = useCallback((value, { name }) => {
    updateFormState(state => {
        return { ...state, [name]: value };
    });
}, []);

<Select onChange={callback} name="test1" />
```

**Reply by itsosmx** [reply]
Thank you ,that will help me a lot

**Comment by hermbo** [comment]
I would just manually write in the name of your select field on your change listener:

```javascript
<Select
   name={'selectName'}
   onChange={(value, action) => handleChange(value, action, 'selectName')}
/>
```

**Reply by itsosmx** [reply]
Thank you,that make my life easier xd
